src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -155,33 +155,40 @@ if (__DEV__) {
       e,
       f,
     ) {
-      ReactErrorUtils._hasCaughtError = false;
-      ReactErrorUtils._caughtError = null;
-
       depth++;
-      const thisDepth = depth;
+
+      let error;
```

Comment:
Ok yeah, I'll do that and also add some comments explaining how this all works, because it's not so straightforward if you don't know what this is all for.